contracts/staking/BaseShareField.sol

```diff
@@ -158,30 +173,29 @@ contract BaseShareField is DAOContract{
         if (totalProductivity != 0 && block.number >= blockStart && blockEnd>= block.number) {
             uint256 multiplier = block.number - lastRewardBlock;
             uint256 reward = multiplier * (rewardsPerBlock * 1e16); // turn it to 18 decimals
-            _accAmountPerShare = _accAmountPerShare + (reward * 1e12 / totalProductivity);
-            uint256 blocksPaid = userInfo.lastRewardTime - userInfo.multiplierResetTime;
-            uint256 blocksPassedFirstMonth = Math.min(ONE_MONTH_IN_BLOCKS,block.number - userInfo.multiplierResetTime); 
-            uint256 blocksToPay = block.number - userInfo.lastRewardTime;
-            uint256 firstMonthBlocksToPay = blocksPaid >= ONE_MONTH_IN_BLOCKS ? 0 : blocksPassedFirstMonth - blocksPaid;
-            uint256 fullBlocksToPay = blocksToPay - firstMonthBlocksToPay;
-            
-           
+            (uint256 blocksToPay, uint256 firstMonthBlocksToPay , uint256 fullBlocksToPay) = _auditCalcs(userInfo);
             
+            _accAmountPerShare = _accAmountPerShare + (reward * 1e12 / totalProductivity);
+            UserInfo memory tempUserInfo = userInfo; // to prevent stacking too deep error any other recommendation? 
             if (blocksToPay != 0){
-                pending = userInfo.amount * _accAmountPerShare / 1e12 - userInfo.rewardDebt;
-                uint rewardPerBlock = pending * 1e12 / blocksToPay / 1e12; // increase resolution
-                pending  = ((firstMonthBlocksToPay * 50 * 1e16) + fullBlocksToPay) * rewardPerBlock / 1e18; // divide 1e18 so reduce it to 18decimals
+                pending = tempUserInfo.amount * _accAmountPerShare / 1e12 - tempUserInfo.rewardDebt;
```

Comment:
why / 1e12

Reply:
This one is old changed with latest commits @sirpy

Reply:
line number 233